### PR TITLE
Deal with empty input on unique-kmers.py

### LINKED ISSUE #1305: unique-kmers.py core dumps on empty input
```
cat < /dev/null | unique-kmers.py -k 20 -
```

yields

```
terminate called without an active exception
Aborted (core dumped)
```

### PR DESCRIPTION
**Pull request by luizirber**
Fixes #1305

## Checklist

- [x] Is it mergeable?
- [x] `make test` Did it pass the tests?
- [ ] `make clean diff-cover` If it introduces new functionality in
  `scripts/` is it tested?
- [x] `make format diff_pylint_report cppcheck doc pydocstyle` Is it well
  formatted?
- [x] Did it change the command-line interface? Only backwards-compatible
  additions are allowed without a major version increment. Changing file
  formats also requires a major version number increment.
- [x] For substantial changes or changes to the command-line interface, is it
  documented in `CHANGELOG.md`? See [keepachangelog](http://keepachangelog.com/)
  for more details.
- [x] Was a spellchecker run on the source code and documentation after
  changes were made?
- [x] Do the changes respect streaming IO? (Are they
  tested for streaming IO?)

### CODE REVIEW

File: lib/hllcounter.cc [hunk 1]
```diff
@@ -424,6 +424,8 @@ void HLLCounter::consume_fasta(
                 // Iterate through the reads and consume their k-mers.
                 try {
                     read = parser->get_next_read();
+                } catch (khmer::StreamReadError) {
+                    break;
```
Comment (luizirber) [inline lib/hllcounter.cc]: I'm just continuing on `StreamError`, but should it stop and report an error? @ctb

**Comment by ctb** [comment]
I don't know!

File: lib/hllcounter.cc (hunk 1 above)
Comment (mr-c) [inline lib/hllcounter.cc]: I would let this be thrown and wrap it into a Python exception in `_khmer.cc`

**Comment by luizirber** [comment]
ready for review and merge @ctb @standage @betatim

File: lib/hllcounter.cc (hunk 1 above)
Comment (standage) [inline lib/hllcounter.cc]: I agree: we probably don't want to be silent about errors in the input data.

**Pull request closed without merging**